pkg/controller/restore_controller.go

```diff
@@ -426,29 +426,6 @@ func (c *restoreController) fetchBackupInfo(backupName string, pluginManager plu
 	}, nil
 }
 
-// fetchFromBackupStorage checks each backup storage location, starting with the default,
-// looking for a backup that matches the given backup name.
-func (c *restoreController) fetchFromBackupStorage(backupName string, pluginManager plugin.Manager) (backupInfo, error) {
-	locations, err := c.backupLocationLister.BackupStorageLocations(c.namespace).List(labels.Everything())
-	if err != nil {
-		return backupInfo{}, errors.WithStack(err)
-	}
-
-	orderedLocations := orderedBackupLocations(locations, c.defaultBackupLocation)
-
-	log := c.logger.WithField("backupName", backupName)
-	for _, location := range orderedLocations {
-		info, err := c.backupInfoForLocation(location, backupName, pluginManager)
-		if err != nil {
-			log.WithField("locationName", location.Name).WithError(err).Error("Unable to fetch backup from object storage location")
-			continue
-		}
-		return info, nil
-	}
-
-	return backupInfo{}, errors.New("not able to fetch from backup storage")
-}
-
 // orderedBackupLocations returns a new slice with the default backup location first (if it exists),
 // followed by the rest of the locations in no particular order.
 func orderedBackupLocations(locations []*api.BackupStorageLocation, defaultLocationName string) []*api.BackupStorageLocation {
```

Comment:
this function is now only used in `backup_sync_controller.go` so I'd probably move it there